command/server.go

```diff
@@ -286,6 +286,17 @@ func (c *ServerCommand) Run(args []string) int {
 		}
 	}
 
+	// If the backend supports service discovery, run service discovery
```

Comment:
Optional minor comment. If this block is around line 225, before core is initialized, I felt that it would be easy to understand the flow. Up to you :)